_pytest/config.py

```diff
@@ -972,11 +972,31 @@ def _initini(self, args):
         ns, unknown_args = self._parser.parse_known_and_unknown_args(args, namespace=self.option.copy())
         r = determine_setup(ns.inifilename, ns.file_or_dir + unknown_args, warnfunc=self.warn)
         self.rootdir, self.inifile, self.inicfg = r
-        self._parser.extra_info['rootdir'] = self.rootdir
-        self._parser.extra_info['inifile'] = self.inifile
-        self.invocation_dir = py.path.local()
         self._parser.addini('addopts', 'extra command line options', 'args')
         self._parser.addini('minversion', 'minimally required pytest version')
+        self._parser.addini('pathtype', 'path implementation to be used', default='pylib')
+
+        self.invocation_dir = self.make_path()
+        self.rootdir = self.make_path(self.rootdir)
```

Comment:
this is why this will be and opt-in and will support a transitional api for quite a while

however py.path has to die

the starting point in any case is compatibility wrappers and the ability to switch
only after all major plugins support the transition apis we can take a look at switching the default

this is something that will have a transition period way beyond the lifetime of python 2